### PR TITLE
feat(typeahead): apply angular watch to the typeahead-editable setting

### PR DESCRIPTION
**Pull request by damonf**
Adds a watch for the typeahead-editable attribute.

### CODE REVIEW

**Comment by icfantv** [comment]
@damonf, what problem is this PR solving?  Also, you're missing functional tests.

Please give [this](https://github.com/angular-ui/bootstrap/blob/master/CONTRIBUTING.md) a read and in the future, it's generally a good idea to run an idea by the team first before spending time on a PR as we don't want you to possibly be wasting your time.  Thanks.

**Comment by wesleycho** [comment]
There was a prior issue reported by @damonf related to this in #4592.

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -44,6 +44,10 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.debounce', 'ui.bootstrap
 
     //should it restrict model values to the ones selected from the popup only?
     var isEditable = originalScope.$eval(attrs.typeaheadEditable) !== false;
+    originalScope.$watch(attrs.typeaheadEditable, function (newVal) {
+      isEditable = newVal !== false;
+      modelCtrl.$setValidity('editable', isEditable);
```
Comment (wesleycho) [inline src/typeahead/typeahead.js]: This is a breaking change, and editable state should not be related to ngModel's validity. This is a bad change.

**Comment by wesleycho** [comment]
This needs tests.

The model $setValidity call needs to be removed as well.

**Comment by damonf** [comment]
Thanks for your patience.
So if i removed the $setValidity call and added some unit tests you would
accept the PR?

**Comment by wesleycho** [comment]
I would be willing to accept this feature with the necessary changes.

**Pull request closed without merging**